tracer/test/test-applications/security/aspnet/Samples.Security.AspNetMvc5/Controllers/IastController.cs

```diff
@@ -22,5 +27,90 @@ public ActionResult WeakHashing(int delay1 = 0, int delay2 = 0)
+                return Content(ToFormattedString(ex));
+            }
+
+            return Content($"No query or username was provided");
+        }
+
+        public static string ToFormattedString(Exception ex)
+        {
+            var message = new StringBuilder();
+            if (ex != null)
+            {
+                message.AppendFormat("Name:       {0}", ex.GetType().FullName);
+                message.Append(Environment.NewLine);
+                message.AppendFormat("Message:    {0}", GetFullMessage(ex));
+                message.Append(Environment.NewLine);
+                message.AppendFormat("StackTrace: {0}", ex.StackTrace);
+                message.Append(Environment.NewLine);
+            }
+            return message.ToString();
+        }
+
+        public static string GetFullMessage(Exception err)
+        {
+            string res = "";
+            while (err != null)
+            {
+                res += err.Message + " ";
+                err = err.InnerException;
+            }
+            return res;
+        }
+
+        public static SQLiteConnection CreateDatabase()
+        {
+            var builder = new SQLiteConnectionStringBuilder();
+            builder.DataSource = ":memory:";
+            var conn = builder.ConnectionString;
+
+            if (string.IsNullOrEmpty(conn))
+            {
+                throw new System.Exception("Cannot create sqlite database.");
+            }
+
+            var dbConnection = new SQLiteConnection(conn);
+            dbConnection.Open();
+            string personCommand = "CREATE TABLE Persons (Id NOT NULL,Name varchar(50) NOT NULL,Surname varchar(50) NULL,Married bit NULL,Gender char(10) NULL,DateOfBirth datetime NULL,Email varchar(50) NULL,FlattedAddress varchar(500) NULL,Mobile varchar(50) NULL,NIF varchar(50) NULL,Phone varchar(50) NULL,PostalCode varchar(50) NULL,Details varchar(5000) NULL,CityIniqueId varchar(50) NULL,CountryUniqueId varchar(50) NULL,ImagePath varchar(500) NULL)";
+            string booksCommand = "CREATE TABLE Books(Id NOT NULL,Title nvarchar(500) NOT NULL,Price numeric(18, 2) NULL,ISBN nvarchar(50) NULL,	Pages int NOT NULL,    PublicationDate datetime NOT NULL,    Author nvarchar(500) NOT NULL,    Editorial nvarchar(50) NULL,	Prologue nvarchar(4000) NULL,	Format int NULL)";
+
+            new SQLiteCommand(booksCommand, dbConnection).ExecuteReader();
+            new SQLiteCommand(personCommand, dbConnection).ExecuteReader();
+            new SQLiteCommand(@"INSERT INTO Persons (Id, Name, Surname, Married, Gender, DateOfBirth, Email, FlattedAddress, Mobile, NIF, Phone, PostalCode, Details, CityIniqueId, CountryUniqueId, ImagePath) VALUES ('D305C1EB-B72E-4340-B5BA-A19D0105A6C2', 'Michael', 'Smith', 0, 'Female    ', CAST(0x00006A1000000000 AS DateTime), 'Michael.Smith@gmail.com', 'Mountain Avenue 55', '650214751', '50099554L', '918084525', '28341', 'Not Defined', NULL, NULL, '~/Images/Antonio.jpg')", dbConnection).ExecuteReader();
```

Comment:
nit: Rather than duplicating all this code between projects, I would be inclined to move it to a helper class and link the file between projects

Reply:
Yes, that is a good option. Done!